File: src/sub/openerror.c
#include <stdbool.h>
#include <stdio.h>

int openerror(const char *subr, const char *file_name, bool single_turbo,
              FILE *logfp) {
    if (subr == NULL) {
        subr = "";
    }
    if (file_name == NULL) {
        file_name = "";
    }

    fprintf(stderr, "WARNING:\n");
    fprintf(stderr, "SUBROUTINE :%s    ERROR OPENING FILE: %s\n", subr,
            file_name);

    if (!single_turbo) {
        FILE *out = (logfp != NULL) ? logfp : stdout;
        fprintf(out, "WARNING:\n");
        fprintf(out, "SUBROUTINE :%s    ERROR OPENING FILE: %s\n", subr,
                file_name);
    }

    return -1;
}

File: src_fortran/sub/openerror.f
c
c
c
c
c
c
      subroutine OPENERROR(subr,char)
c
      implicit none
      include '../inc/vel_com.inc'
c
      character*(*) subr,char
      write(6,*)'WARNING:'
      write(6,*)'SUBROUTINE :',subr,'    ERROR OPENING FILE: ',char
      if(.not.single_turbo)then
         write(16,*)'WARNING:'
         write(16,*)'SUBROUTINE :',subr,'    ERROR OPENING FILE: ',char
      endif
      stop'Openerror; program VELEST stopped !'
      end ! of subr. openerror
